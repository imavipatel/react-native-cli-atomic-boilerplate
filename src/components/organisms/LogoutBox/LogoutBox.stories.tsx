import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { LogoutBox } from './LogoutBox';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

storiesOf('LogoutBox', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => (
    <LogoutBox onPressLogout={() => {}}>Logout Box Title</LogoutBox>
  ));
