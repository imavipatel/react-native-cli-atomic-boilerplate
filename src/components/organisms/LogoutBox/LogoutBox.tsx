import { PressableLabel } from 'components/atoms';
import React, { ReactNode } from 'react';
import { View } from 'react-native';

import { styles } from './LogoutBox.style';

interface LogoutBoxProps {
  children: ReactNode;
  onPressLogout: () => void;
  testID: string;
}

export const LogoutBox: React.FC<LogoutBoxProps> = props => {
  return (
    <View style={styles.container}>
      <PressableLabel onPress={props.onPressLogout} testID={props.testID}>
        {props.children}
      </PressableLabel>
    </View>
  );
};
