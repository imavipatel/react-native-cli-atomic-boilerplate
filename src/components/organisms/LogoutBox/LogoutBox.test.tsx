import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { LogoutBox } from './LogoutBox';

describe('LogoutBox', () => {
  const spyOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(
      <LogoutBox onPressLogout={spyOnPress} testID="logout-label">
        Logout Box title
      </LogoutBox>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should call the provided onPress function when pressed', () => {
    const component = (
      <LogoutBox onPressLogout={spyOnPress} testID="logout-label">
        Logout Box title
      </LogoutBox>
    );
    const wrapper = render(component);

    fireEvent(wrapper.getByTestId('logout-label'), 'onPress');
    expect(spyOnPress).toHaveBeenCalledTimes(1);
  });
});
