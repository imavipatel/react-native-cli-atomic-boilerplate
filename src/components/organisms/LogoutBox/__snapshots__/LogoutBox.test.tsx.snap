// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutBox should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 2,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="logout-label"
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      testID="label"
    >
      Logout Box title
    </Text>
  </View>
</View>
`;
